tartufo/util.py

```diff
@@ -242,7 +257,7 @@ def _style_func(msg: str, *_: Any, **__: Any) -> str:
     style_ok = style_error = style_warning = partial(_style_func)
 
 
-def fail(msg: str, ctx: click.Context, code: int = 1) -> NoReturn:
+def fail(msg: str, ctx: click.Context, code: int = 1) -> None:
```

Comment:
i was getting a `mypy` failure on the pre-commit hook with:

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1
tartufo/util.py:261: error: Implicit return in function which does not return
```
i can try reverting this to see if it complains in CICD. this change as is has caused another side-effect linting issue, which i was surprised wasn't flagged locally in pre-commit

Reply:
I think the mypy failure you're getting now is due to the return-value change:

```
tartufo/commands/update_signatures.py:273: error: Incompatible return value type (got "Optional[GitRepoScanner]", expected "GitRepoScanner")  [return-value]
```
It used to be that `scanner` was known not to be `None` by L273 because of the test at L250 combined with the knowledge that control would not return from the call to `util.fail()` on L252. Now mypy assumes control will return and that consequently `scanner` could still be `None` -- thus the error.

Reply:
yup, exactly. was just posting the previous mypy failure which led me to the `NoReturn` -> `None` change. wrapping up wordsmithing the documentation change and i'll push up a revert for this line along with it.